libvast/vast/bloom_filter.hpp

```diff
@@ -110,7 +110,11 @@ class bloom_filter : detail::equality_comparable<
 
   template <class Inspector>
   friend auto inspect(Inspector& f, bloom_filter& x) {
-    return f(caf::meta::type_name("bloom_filter"), x.hasher_, x.bits_);
+    auto result = f(caf::meta::type_name("bloom_filter"), x.hasher_, x.bits_);
+    // When deserializing into a vector that was already bigger than required,
+    // CAF will reuse the storage but not release the excess afterwards.
+    x.bits_.shrink_to_fit();
+    return result;
```

Comment:
I think this should only be done when loading.

```suggestion
    auto& bits = x.bits_;
    auto load = [&]() -> caf::error {
      // When deserializing into a vector that was already bigger than required,
      // CAF will reuse the storage but not release the excess afterwards.
      bits_.shrink_to_fit();
      return caf::none;
    };
    auto result = f(caf::meta::type_name("bloom_filter"), x.hasher_, bits, load);
    return result;
```

And add `#include <caf/meta/load_callback.hpp>` at the top.

Reply:
Right idea, but the suggested code looks wrong.

Reply:
Sounds like we want to if-constexpr on `Inspector::writes_state`?

Reply:
Ah, the suggested code missed wrapping the lambda in `caf::meta::load_callback`. 

@mavam That's how the load_callback is implemented.